### PR TITLE
[kube-prometheus-stack] : allow to customize alertmanager name to avoir name > 63 chars

### PR DESCRIPTION
**Pull request by survivant**
the goal of this PR, is to fix the issue : https://github.com/prometheus-community/helm-charts/issues/437

@vsliouniaev @bismarck @gianrubio @gkarthiks @scottrigby @Xtigyro

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by gkarthiks — COMMENTED** [review]
@survivant thanks for the PR. Please sign the DCO and you might need to bump the chart version as well.

**Comment by survivant** [comment]
@gkarthiks wierd, I did the signoff few times but the check is still not detected

File: charts/kube-prometheus-stack/Chart.yaml
Comment (gkarthiks) [inline charts/kube-prometheus-stack/Chart.yaml:21]: seems you might need to bump the chart version as well.

Comment (survivant) [reply charts/kube-prometheus-stack/Chart.yaml]: bump version to resolve conflit

**Review by zanhsieh — APPROVED** [review]

**Comment by survivant** [comment]
@zanhsieh can you merge this PR ?

**Comment by bismarck** [comment]
@survivant Linter is failing and you need to resolve conflicts.

**Comment by survivant** [comment]
there is someone that can take a look at the errors ?

Error: Error installing charts: Error processing charts
 ✖︎ kube-prometheus-stack => (version: "12.9.3", path: "charts/kube-prometheus-stack") > Error waiting for process: exit status 1
 ✔︎ prometheus => (version: "13.0.3", path: "charts/prometheus")

don't know how that part works.

**Comment by bismarck** [comment]
See: https://github.com/prometheus-community/helm-charts/pull/516/checks?check_run_id=1644115529#step:6:197

Try installing the chart with this values file: https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/ci/02-test-without-crds-values.yaml

**Comment by survivant** [comment]
good, I can reproduce the issue locally
$ helm install test-kps . -f ci/02-test-without-crds-values.yaml
Error: rendered manifests contain a resource that already exists. Unable to continue with install: could not get information about the resource: invalid resource name "test-kps-kube-prometheus-s-%!s(MISSING)": [may not contain '%']

I'll take a look

**Comment by survivant** [comment]
@bismarck this time all the checks are Green 👍

**Review by zanhsieh — DISMISSED** [review]

**Comment by survivant** [comment]
almost merged :)

@bismarck, @gianrubio, @gkarthiks, @scottrigby, @vsliouniaev, @Xtigyro

**Review by monotek — APPROVED** [review]

**Comment by survivant** [comment]
why the merge is blocked ?

**Comment by monotek** [comment]
Needs an approval of one of the chart maintainers too.

**Comment by survivant** [comment]
do you know who we could ping ?

**Comment by monotek** [comment]
https://github.com/prometheus-community/helm-charts/blob/main/.github/CODEOWNERS#L10

**Comment by survivant** [comment]
ok, thanks.  They are already in the loop

**Comment by gkarthiks** [comment]
hi @survivant one last rebase for resolving the conflicts pls?

**Comment by survivant** [comment]
@gkarthiks should be ok now

**Review by gkarthiks — DISMISSED** [review]
I am good with the kube-prom-stack. @monotek  or someone needs to approve the changes in the `charts/prometheus`

**Comment by gkarthiks** [comment]
Since you have changes in `prometheus` charts as well, you need approval from one of https://github.com/prometheus-community/helm-charts/blob/main/.github/CODEOWNERS#L11 too.

**Comment by zanhsieh** [comment]
@survivant 
Could you bump the chart version to latest again please?

**Comment by survivant** [comment]
@zanhsieh , I'll wait after someone from the CodeOwners to take a look, because I don't want to maintain that PR everyday if that's OK with you

**Review by zanhsieh — DISMISSED** [review]

**Comment by zanhsieh** [comment]
@survivant 
I approved. Just fix DCO yourself.

**Review by bismarck — CHANGES_REQUESTED** [review]
@survivant Could you revert your changes to charts/prometheus/Chart.yaml ? You can make changes to one chart at a time and you have no changes for the prometheus chart.

File: charts/prometheus/Chart.yaml
Comment (survivant) [reply charts/prometheus/Chart.yaml]: @bismarck  cool, thanks for the feedback.  I just keep bumping the version to avoid conflit because the PR is almost a month old.. I didn't have conflit back then.  

Just tell me which version I should put there and I'll do it

Comment (bismarck) [reply charts/prometheus/Chart.yaml]: @survivant just revert the changes to this file. It isn't relevant to your PR.

Comment (survivant) [reply charts/prometheus/Chart.yaml]: should be ok @bismarck

**Comment by survivant** [comment]
@bismarck conflit resolved, but look like I'm not able to signoff the DCO now :(

**Comment by survivant** [comment]
@bismarck can I codeOwner take the lead of this PR.  The more we wait to merge more conflits appears.

**Comment by bismarck** [comment]
@survivant Need to resolve conflicts again. You also need to sign your commits (that's why the DCO check is failing).

**Comment by monotek** [comment]
@survivant  
To solve the DCO check see: https://github.com/prometheus-community/helm-charts/pull/516/checks?check_run_id=1724900309

**Comment by survivant** [comment]
> @survivant
> To solve the DCO check see: https://github.com/prometheus-community/helm-charts/pull/516/checks?check_run_id=1724900309

@monotek 
yes I know.  but it took too long to merge that PR that there are lot of conflits and the tests start failing again :(  I don't have time anymore to fix that.  

I'll check when I'll have time to do a new PR from scratch.

**Comment by gkarthiks** [comment]
It would be easy if we have a separate PR for every chart that needs change rather combining them altogether.

**Comment by vsliouniaev** [comment]
Overall this seems like a very inconsistent change with the rest of the available configs in the chart. The length of names created by the prometheus operator PVs will be even longer.

**Comment by survivant** [comment]
the changes works.  The version that I put is the version that we are using right now.  It fixed the PVC that were too long.

**Comment by vsliouniaev** [comment]
I understand the fix works, but this a unique change to just one part of the chart and doesn't address the larger issue with long resource names.

It is not necessary to make this change since it is possible to give the overall chart a shorter name, so I don't see the point in adding this feature.

**Comment by survivant** [comment]
The point was to allow people to override the default names to be able to generate shorter names. Right now, it's impossible to get the alertmanager working with persistence enabled.  I see benefits to allow that.  I don't see the point of using a chart that it's not working.  

PS.  I won't continue to update this PR.  I'll just stick with my version that works in my cluster until a switch to another chart that work (without a hack) with persistance enabled.

**Pull request closed without merging**
